examples/chip-tool/templates/commands.zapt

```diff
@@ -2,23 +2,224 @@
 
 #pragma once
 
+#include <charconv>
 #include <cstdint>
+#include <functional>
 #include <string>
 #include <type_traits>
+#include <vector>
 
 #include <app-common/zap-generated/cluster-objects.h>
 #include <app/data-model/DecodableList.h>
 #include <app/data-model/Nullable.h>
 #include <commands/clusters/ModelCommand.h>
 #include <commands/common/CommandInvoker.h>
+#include <commands/common/ValueParser.h>
 #include <lib/core/CHIPSafeCasts.h>
 #include <lib/support/BytesToHex.h>
 #include <lib/support/Span.h>
 #include <lib/support/TypeTraits.h>
 #include <zap-generated/CHIPClientCallbacks.h>
 #include <zap-generated/CHIPClusters.h>
 
-// Value logging functions.  The non-generated ones depend on the
+namespace chip {
+
+{{#zcl_clusters}}
+{{#zcl_structs}}
+CHIP_ERROR ParseValue(const char *& iter, const char * end, {{zapTypeToEncodableClusterObjectType name ns=parent.name}} & value,
+                      std::vector<std::function<void(void)>> & freeFunctions);
+{{/zcl_structs}}
+{{/zcl_clusters}}
+
+template <typename T>
+CHIP_ERROR ParseValue(const char *& iter, const char * end, chip::app::DataModel::Nullable<T> & value,
```

Comment:
the nullable/optional/bitflags and such seem to not depend on template structs/types. Should be moved to separate header as well.